src/cow.jl

```diff
@@ -0,0 +1,141 @@
+using IRTools
+using MacroTools
+
+using IRTools: BasicBlock, stmt, var
+
+const MUTATING_OPS = (:push!, :pop!, :setindex!)
+
+function copy_object(obj)
+    ct = _current_task()
+    ct.storage == nothing && (ct.storage = IdDict())
```

Comment:
Often it's better to compare `nothing` (and singletons in general) with `===`:
```suggestion
    ct.storage === nothing && (ct.storage = IdDict())
```